internal/provider/kubernetes/controller.go

```diff
@@ -1435,6 +1440,7 @@ func (r *gatewayAPIReconciler) watchResources(ctx context.Context, mgr manager.M
 		if err := c.Watch(
 			source.Kind(mgr.GetCache(), &mcsapi.ServiceImport{}),
 			handler.EnqueueRequestsFromMapFunc(r.enqueueClass),
+			predicate.GenerationChangedPredicate{},
```

Comment:
I think you missed `EndpointSlice`, can you please also add the predicate for `EndpointSlice` ?